client/beefy/src/worker.rs

```diff
@@ -500,19 +500,27 @@ where
 
 			self.on_demand_justifications.cancel_requests_older_than(block_num);
 
-			if let Err(e) = self.backend.append_justification(
-				BlockId::Number(block_num),
-				(BEEFY_ENGINE_ID, finality_proof.encode()),
-			) {
-				error!(target: "beefy", "🥩 Error {:?} on appending justification: {:?}", e, finality_proof);
-			}
+			let maybe_block_hash =
+				self.backend.blockchain().expect_block_hash_from_id(&BlockId::Number(block_num));
```

Comment:
```suggestion
			let block_hash =
				match self.backend.blockchain().expect_block_hash_from_id(&BlockId::Number(block_num)) {
				Ok(),
				Err(),
			}
```

Would make the code a little bit easier.

Reply:
reworked in [4930349](https://github.com/paritytech/substrate/pull/12551/commits/4930349a6e97a7d96564da1cb95459ca942d5e79)